### ISSUE TITLE
The number of Kafka is incorrect

### ISSUE THREAD

**Issue by wu-sheng** [apache/incubator-skywalking#1177]
![image](https://user-images.githubusercontent.com/5441976/39664274-5d476350-50b3-11e8-84ca-85243c9becaa.png)
 
In this topology, these is only one Kafka, and same in reality env.

**Comment by wu-sheng** [comment]
I have pinpointed this issue. We really need to refact all component libraries. This is an agent issue, caused by old design. @peng-yongsheng  @ascrutae

**Comment by ajanthan** [comment]
@wu-sheng I guess you have found the root cause of the issue. Could you please describe your findings?

**Comment by wu-sheng** [comment]
@ajanthan Look like you are facing this issue too. I can explain to you with more details. The reason I called this a design issue, it is in old days(3.x series), we have no such a powerful UI, just topology. Then in order to visualize conjecture nodes(such as: MySQL, Kafka, which would be installed an agent), we put the client library names as the server name. 

For example, we set component name of Kafka consumer as `Kafka`, only, which triggers the issue described in here. 

So we have to change this, and put the real component names in agent, then do component->server-side mapping for the UI. 

See https://github.com/apache/incubator-skywalking/pull/1266 , that includes all the changes for this issue. But sadly, today's travis-ci has some network issues... fails all the time... can't merge it.
